packages/toolpad-app/src/runtime/ToolpadApp.tsx

```diff
@@ -292,8 +292,9 @@ function RenderedNodeContent({ node, childNodeGroups, Component }: RenderedNodeC
 
   const layoutElementProps = React.useMemo(() => {
     if (appDom.isElement(node) && isPageRow(node)) {
+      const children = childNodeGroups.children || [];
       return {
-        layoutColumnSizes: childNodeGroups.children.map((child) => child.layout?.columnSize?.value),
+        layoutColumnSizes: children.map((child) => child.layout?.columnSize?.value),
```

Comment:
@apedroferreira I managed to reproduce the crash locally and it seems it was caused because `children` was undefined on `childNodeGroups`. If I default to empty array here it no longer crashes but something else weird happens and I'm not sure yet why exactly:

<img width="447" alt="image" src="https://user-images.githubusercontent.com/437214/199223822-1a0caa7c-8b05-406f-adec-ca470edb26d1.png">

Upon undo/redo I get this weird row element added with `Drop component here` text. I can actually select it and delete it but it shouldn't be here. Do you have any idea why this might be happening and whether it can be related to this change to `layoutColumnsizes` calculation?

Reply:
Maybe the throttling is not enough to stop some in-between state where rows actually don't have `children` or the `children` prop.
Does changing every `appDom.createElement(dom, PAGE_ROW_COMPONENT_ID, {})` in the `RenderOverlay` file to `appDom.createElement(dom, PAGE_ROW_COMPONENT_ID, { children: [] })` maybe fix the issue?

Reply:
My guess is that it has to do with the same reason it failed for selected nodes - when you remove something, code expects that it's there, just as you said - it should never be `undefined`, but upon undo action it probably will be `undefined`

Reply:
Turns out throttling is just not working as intended. I proposed a fix here: https://github.com/mui/mui-toolpad/pull/1266

Reply:
Thanks a lot! Yeah that does seem to solve that issue. It's a little bit weird because even with throttle implementation it undo/redo itself worked (in terms of getting me to previous or next state) but as you pointed out it's possible that it created unwanted intermediate state which lead to error thrown 🤔 

Anyways, thanks a lot for looking into it and helping out ❤️

Reply:
I think there's a misunderstanding about what `useThrottled` does. It throttles updates to a value, it doesn't give you a function that wraps another function with throttling.

There are 4 hooks:

* `useDebounced`: debounces a value
* `useDebouncedHandler`: wraps a function with debounce behavior
* `useThrottled`: throttles a value
* `useThrottledHandler`: wraps a function with throttling behavior (doesn't exist yet in the codebase)

It seems like you were looking for `useThrottledHandler`, and you used something analog to `useDebouncedHandler`

Reply:
I just checked but it seems we don't even have `useThrottledHandler` 🤔
IMO this is a good illustration of the problem I mentioned in the comment below: https://github.com/mui/mui-toolpad/pull/1225#discussion_r1012682888

1. It's not really clear whether the implementation that we have is bug free?
2. It's more difficult to know that these methods even exist in our codebase
3. It's hard to know what exactly should be used as there are no docs besides the types themselves